auv_control/auv_control/scripts/reference_pose_publisher.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+import rospy
+from nav_msgs.msg import Odometry
+from geometry_msgs.msg import Pose, Twist
+from auv_msgs.srv import SetDepth, SetDepthRequest, SetDepthResponse
+from tf.transformations import quaternion_from_euler, euler_from_quaternion
+from auv_common_lib.control.enable_state import ControlEnableHandler
+
+
+class DepthControllerNode:
+    def __init__(self):
+        rospy.init_node("reference_pose_publisher_node")
+
+        # Initialize subscribers
+        self.odometry_sub = rospy.Subscriber(
+            "/taluy/odometry", Odometry, self.odometry_callback
+        )
+        self.set_depth_service = rospy.Service(
+            "/taluy/set_depth", SetDepth, self.target_depth_handler
+        )
+        self.cmd_vel_sub = rospy.Subscriber(
+            "/taluy/cmd_vel", Twist, self.cmd_vel_callback
+        )
+
+        # Initialize publisher
+        self.cmd_pose_pub = rospy.Publisher("/taluy/cmd_pose", Pose, queue_size=10)
+
+        self.control_enable_handler = ControlEnableHandler(1.0)
+
+        # Initialize internal state
+        self.target_depth = 0.0
+        self.target_heading = 0.0
+        self.last_cmd_time = rospy.Time.now()
+
+        # Parameters
+        self.update_rate = rospy.get_param("~update_rate", 10)
+
+        rospy.Timer(rospy.Duration(1.0 / self.update_rate), self.control_loop)
+
+    def target_depth_handler(self, req: SetDepthRequest) -> SetDepthResponse:
+        self.target_depth = req.target_depth
+        return SetDepthResponse(
+            success=True, message=f"Target depth set to {self.target_depth}"
+        )
+
+    def odometry_callback(self, msg):
+        if self.control_enable_handler.is_enabled():
+            return
+
+        quaterion = [
+            msg.pose.pose.orientation.x,
+            msg.pose.pose.orientation.y,
+            msg.pose.pose.orientation.z,
+            msg.pose.pose.orientation.w,
+        ]
+        _, _, self.target_heading = euler_from_quaternion(quaterion)
+
+    def cmd_vel_callback(self, msg):
+        if not self.control_enable_handler.is_enabled():
+            return
+
+        dt = (rospy.Time.now() - self.last_cmd_time).to_sec()
+        dt = min(dt, 0.1)
```

Comment:
Can you move 0.1 to init as param?

This can be named
command_timeout, which later needs to be sync's with other nodes just like "control_rate" param.